accounts/abi/bind/backends/simulated.go

```diff
@@ -349,8 +348,11 @@ func (b *SimulatedBackend) CallContract(ctx context.Context, call ethereum.CallM
 	if err != nil {
 		return nil, err
 	}
-	rval, _, _, err := b.callContract(ctx, call, b.blockchain.CurrentBlock(), state)
-	return rval, err
+	res, err := b.callContract(ctx, call, b.blockchain.CurrentBlock(), state)
+	if err != nil {
+		return nil, err
```

Comment:
I am not 100% sure, but I think we could return the reason for failing here too.
e.g. 
```
if res.Failed() {
	errMsg := fmt.Sprintf("transaction reverted (%x)", result.Err)
	if len(res.Revert()) > 0 {
		errMsg += fmt.Sprintf(" (0x%x)", result.Revert())
	}
	return nil, errors.New(errMsg)
}